opentelemetry-sdk/src/testing/trace/span_exporters.rs

```diff
@@ -0,0 +1,168 @@
+use crate::{
+    export::{
+        trace::{ExportResult, SpanData, SpanExporter},
+        ExportError,
+    },
+    trace::{Config, SpanEvents, SpanLinks},
+    InstrumentationLibrary,
+};
+use async_trait::async_trait;
+use crossbeam_channel::{unbounded, Receiver, SendError, Sender};
+use futures_util::future::BoxFuture;
+pub use opentelemetry::testing::trace::TestSpan;
+use opentelemetry::trace::{
+    SpanContext, SpanId, SpanKind, Status, TraceFlags, TraceId, TraceState,
+};
+use std::fmt::{Display, Formatter};
+
+pub fn new_test_export_span_data() -> SpanData {
+    let config = Config::default();
+    SpanData {
+        span_context: SpanContext::new(
+            TraceId::from_u128(1),
+            SpanId::from_u64(1),
+            TraceFlags::SAMPLED,
+            false,
+            TraceState::default(),
+        ),
+        parent_span_id: SpanId::INVALID,
+        span_kind: SpanKind::Internal,
+        name: "opentelemetry".into(),
+        start_time: opentelemetry::time::now(),
+        end_time: opentelemetry::time::now(),
+        attributes: Vec::new(),
+        dropped_attributes_count: 0,
+        events: SpanEvents::default(),
+        links: SpanLinks::default(),
+        status: Status::Unset,
+        resource: config.resource,
+        instrumentation_lib: InstrumentationLibrary::default(),
+    }
+}
+
+#[derive(Debug)]
+pub struct TestSpanExporter {
```

Comment:
Can we add some documents for `TestSpanExporter`? If we are making it public

Reply:
This is not made public. One InMemory*Exporters are public.

Reply:
Ah I see